constants/ram_constants.asm

```diff
@@ -0,0 +1,120 @@
+; wStrengthFlags::
+	const_def
+	const BIT_SEEN_BY_TRAINER      ; 0
+	const BIT_BOULDER_DUST         ; 1
+	const BIT_TURNING              ; 2
+	const BIT_USING_GENERIC_PC     ; 3
+	const BIT_NO_SPRITE_UPDATES    ; 4
+	const BIT_NO_MENU_BUTTON_SOUND ; 5
+	const BIT_TRIED_PUSH_BOULDER   ; 6
+	const BIT_PUSHED_BOULDER       ; 7
+
+; wOptions::
+DEF TEXT_DELAY_MASK EQU %111
+	const_def 6
+	const BIT_BATTLE_SHIFT     ; 6
+	const BIT_BATTLE_ANIMATION ; 7
+
+DEF TEXT_DELAY_FAST   EQU %001 ; 1
+DEF TEXT_DELAY_MEDIUM EQU %011 ; 3
+DEF TEXT_DELAY_SLOW   EQU %101 ; 5
+
+; wLetterPrintingDelayFlags::
+	const_def
+	const BIT_FAST_TEXT_DELAY ; 0
+	const BIT_TEXT_DELAY      ; 1
+
+; wObtainedBadges::
+; wBeatGymFlags::
+	const_def
+	const BIT_BOULDERBADGE ; 0
+	const BIT_CASCADEBADGE ; 1
+	const BIT_THUNDERBADGE ; 2
+	const BIT_RAINBOWBADGE ; 3
+	const BIT_SOULBADGE    ; 4
+	const BIT_MARSHBADGE   ; 5
+	const BIT_VOLCANOBADGE ; 6
+	const BIT_EARTHBADGE   ; 7
+DEF NUM_BADGES EQU const_value
+
+; wStatusFlags::
+	const_def
+	const BIT_STRENGTH_ACTIVE           ; 0
+	const BIT_SURF_ALLOWED              ; 1
+	const_skip                          ; 2 ; unused
+	const BIT_GOT_OLD_ROD               ; 3
+	const BIT_GOT_GOOD_ROD              ; 4
+	const BIT_GOT_SUPER_ROD             ; 5
+	const BIT_GAVE_SAFFRON_GUARDS_DRINK ; 6
+	const BIT_UNUSED_CARD_KEY           ; 7
+
+; wStatusFlags2::
+	const_def
+	const BIT_WILD_ENCOUNTER_COOLDOWN ; 0
+	const BIT_NO_AUDIO_FADE_OUT       ; 1
+
+; wStatusFlags3::
+	const_def
+	const BIT_INIT_TRADE_CENTER_FACING ; 0
+	const_skip 2                       ; 1-2 ; unused
+	const BIT_WARP_FROM_CUR_SCRIPT     ; 3
+	const BIT_ON_DUNGEON_WARP          ; 4
+	const BIT_NO_NPC_FACE_PLAYER       ; 5
+	const BIT_UNKNOWN_3_6              ; 6
+	const BIT_UNKNOWN_3_7              ; 7
```

Comment:
```suggestion
; wStatusFlags3::
	const_def
	const BIT_INIT_TRADE_CENTER_FACING ; 0
	const_skip 2                       ; 1-2 ; unused
	const BIT_WARP_FROM_CUR_SCRIPT     ; 3
	const BIT_ON_DUNGEON_WARP          ; 4
	const BIT_NO_NPC_FACE_PLAYER       ; 5
	const BIT_IN_TRAINER_BATTLE        ; 6
	const BIT_PRINT_END_BATTLE_TEXT    ; 7
```

Reply:
Not 100% on the `IN_TRAINER_BATTLE` one, but the `PRINT_END_BATTLE_TEXT` should be correct.

Reply:
`BIT_UNKNOWN_3_6` is set when you talk to a trainer yourself. (It's also set when one sees you, but only after they've walked up and talked to you. `BIT_SEEN_BY_TRAINER` is set while they saw you and are walking; you stop moving down Cycling Road while that's on.)